library/management/command/src/main/java/org/quiltmc/qsl/command/api/EntitySelectorOptionRegistry.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.command.api;
+
+import java.util.function.Predicate;
+
+import org.jetbrains.annotations.NotNull;
+
+import net.minecraft.command.EntitySelectorOptions;
+import net.minecraft.command.EntitySelectorReader;
+import net.minecraft.text.Text;
+import net.minecraft.util.Identifier;
+
+import org.quiltmc.qsl.command.mixin.EntitySelectorOptionsAccessor;
+
+/**
+ * Class to allow registration of custom {@link net.minecraft.command.EntitySelectorOptions entity selector options}.
+ *
+ * <p>These are registered with namespaced IDs to avoid name collisions. These are then converted to a name of the form
+ * "namespace_path". Due to {@link com.mojang.brigadier.StringReader#isAllowedInUnquotedString(char) limitations} in how
+ * entity selectors may be named, the character "{@code /}" is replaced with "{@code _}".</p>
```

Comment:
```suggestion
 * <p>
 * These are registered with namespaced IDs to avoid name collisions. These are then converted to a name of the form
 * "namespace_path". Due to {@link com.mojang.brigadier.StringReader#isAllowedInUnquotedString(char) limitations} in how
 * entity selectors may be named, the character "{@code /}" is replaced with "{@code _}".
```